### ISSUE TITLE
[web worker / ngc] Aot compilation error in web worker scenario

### ISSUE THREAD

**Issue by mdaveynis** [angular/angular#12097]
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Change #10334 from `selectTarget[member]` to  `simplify(selectTarget[member])` in modules/@angular/compiler-cli/src/static_reflector.ts broke ngc in web worker scenario with error below.

`Error: parameters received {"filePath":"c:/Development/angular_quickstart/node_modules/@angular/platform-browser/src/security/dom_sanitization_service.d.ts","name":"DomSanitiz
erImpl"} which is not a StaticSymbol                                                                                                                                           
    at StaticReflector.parameters (c:\Development\angular_quickstart\node_modules\@angular\compiler-cli\src\static_reflector.js:92:19)                                         
    at CompileMetadataResolver.getDependenciesMetadata (c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14317:56)                     
    at CompileMetadataResolver.getTypeMetadata (c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14282:28)                             
    at CompileMetadataResolver.getProviderMetadata (c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14473:42)                         
    at c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14421:47                                                                       
    at Array.forEach (native)                                                                                                                                                  
    at CompileMetadataResolver.getProvidersMetadata (c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14405:21)                        
    at c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14412:43                                                                       
    at Array.forEach (native)                                                                                                                                                  
    at CompileMetadataResolver.getProvidersMetadata (c:\Development\angular_quickstart\node_modules\@angular\compiler\bundles\compiler.umd.js:14405:21)                        
Compilation failed`

**Expected behavior**
No error

**Minimal reproduction of the problem with instructions**

Using https://github.com/mdaveynis/angular_quickstart repository master branch either run "npm start" or "npm run ngc"

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**
Windows 10, VSCode
- **Angular version:** 2.0.1
  
  
- **Browser:** N/A
- **Language:** TypeScript 2.0.3
- **Node (for AoT issues):** `node --version` =   5.3.0

**Comment by steve8708** [comment]
Looks to be same issue as #11276

**Comment by steve8708** [comment]
PR to resolve this sent #12158

**Comment by mhevery** [comment]
Fixed by #12158
